kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/partitioner/AvroFieldValuePartitionerConfig.java

```diff
@@ -0,0 +1,133 @@
+package org.hypertrace.core.kafkastreams.framework.partitioner;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.Maps;
+import com.google.common.util.concurrent.AtomicDouble;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import lombok.Value;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@NoArgsConstructor
+@Getter
+class AvroFieldValuePartitionerConfig {
+  private static final Splitter SPLITTER = Splitter.on(",").omitEmptyStrings().trimResults();
+
+  static final String PARTITIONER_CONFIG_PREFIX = "avro.field.value.partitioner";
+
+  static final String TOPICS_CONFIG_PREFIX = "topics";
+  static final String FIELD_NAME = "field.name";
+  static final String EXCLUDED_PARTITIONS = "excluded.partitions";
+  static final String DEFAULT_GROUP_WEIGHT = "default.group.weight";
+
+  static final String GROUPS_CONFIG_PREFIX = "groups";
+  static final String GROUP_MEMBERS = "members";
+  static final String GROUP_WEIGHT = "weight";
+
+  Map<String, String> fieldNameByTopic;
+  Map<String, Set<Integer>> excludedPartitionsByTopic;
+  PartitionGroupConfig defaultGroupConfig;
+  Map<String, PartitionGroupConfig> groupConfigByMember;
+
+  @Value
+  @ToString
+  static class PartitionGroupConfig {
+    // This represents some range between 0-1 (e.g. 0.6-0.8) that specifies this group's share
+    double normalizedFractionalStart;
+    double normalizedFractionalEnd;
+  }
+
+  public void configure(Map<String, ?> streamConfigMap) {
+    this.fieldNameByTopic = Maps.newHashMap();
+    this.excludedPartitionsByTopic = Maps.newHashMap();
+
+    Map<String, Object> partitionerConfigProps =
+        streamConfigMap.entrySet().stream()
+            .filter(entry -> entry.getKey().startsWith(PARTITIONER_CONFIG_PREFIX))
```

Comment:
Why is this needed when we just `getConfig` the prefix later and drop the other stuff?

Reply:
the map is not directly parseable with ConfigFactory. Hence, the parsing